CRM/Event/BAO/Participant.php

```diff
@@ -1904,4 +1904,28 @@ public static function getSelfServiceEligibility(int $participantId, string $url
     return $details;
   }
 
+  /**
+   * Callback for hook_civicrm_pre().
+   * @param \Civi\Core\Event\PreEvent $event
+   * @throws CRM_Core_Exception
+   */
+  public static function self_hook_civicrm_pre(\Civi\Core\Event\PreEvent $event) {
```

Comment:
@mattwire this function is excellent. Can you put up a PR just with this hook (renamed to the new field name of course) with a unit test.